receiver/signozkafkareceiver/kafka_receiver.go

```diff
@@ -400,7 +484,50 @@ func (c *kafkaLogsConsumer) Shutdown(context.Context) error {
 	return c.consumerGroup.Close()
 }
 
+type baseConsumerGroupHandler struct {
+	logger          *zap.Logger
+	retryInterval   time.Duration
+	pausePartition  chan int32
+	resumePartition chan int32
+}
+
+// wrap is now a method of BaseConsumerGroupHandler
+func (b *baseConsumerGroupHandler) WithMemoryLimiter(ctx context.Context, claim sarama.ConsumerGroupClaim, consume func() error) error {
+	// Execute f() immediately
+	err := consume()
+	if err == nil || !(err.Error() == errHighMemoryUsage.Error()) {
+		return err
+	}
+
+	// If errHighMemoryUsage is encountered, enter the retry loop
+	ticker := time.NewTicker(b.retryInterval)
+	defer ticker.Stop()
+
+	b.logger.Info("applying initial backpressure on Kafka due to high memory usage", zap.Int32("partition", claim.Partition()), zap.String("topic", claim.Topic()))
+	b.pausePartition <- claim.Partition()
```

Comment:
With the last commit of resuming and pausing all, the channel is no longer holding the partition but just a way to signal it. Please update it struct{}{} or something so that future readers won't get confused.

Reply:
Resolved!